src/ui/Facet/FacetSearch.ts

```diff
@@ -35,22 +31,19 @@ export class FacetSearch {
   public searchResults: HTMLElement;
   public search: HTMLElement;
 
-  private magnifier: HTMLElement;
-  private wait: HTMLElement;
   private clear: HTMLElement;
-  private middle: HTMLElement;
   private input: HTMLInputElement;
   private facetSearchTimeout: number;
   private facetSearchPromise: Promise<IIndexFieldValue[]>;
   private moreValuesToFetch = true;
   private onResize: (...args: any[]) => void;
   private onDocumentClick: (e: Event) => void;
-  private searchBarIsAnimating: boolean = false;
   private lastSearchWasEmpty = true;
+  private facetSearchElement: FacetSearchElement;
 
   constructor(public facet: Facet, public facetSearchValuesListKlass: IFacetSearchValuesListKlass, private root: HTMLElement) {
-    this.searchResults = document.createElement('ul');
-    $$(this.searchResults).addClass('coveo-facet-search-results');
+    this.facetSearchElement = new FacetSearchElement();
+    this.searchResults = this.facetSearchElement.searchResults;
```

Comment:
Could storing results like this be problematic when facet search results update? You will need to maintain `this.searchResults`? If so, perhaps a getter would be better?

```
private get searchResults() {...}
```

Reply:
Good point, I will use a getter. `searchResults` is never re-assigned but it is safer to use a getter.